test/jasmine/tests/parcoords_test.js

```diff
@@ -868,7 +871,65 @@ describe('parcoords basic use', function() {
 
     });
 
-    it('@gl Calling `Plotly.restyle` with a string path should amend the preexisting parcoords', function(done) {
+    function _getGrayRatio(secondPass, msg)
+        var canvases = d3.selectAll('.gl-canvas');
+        expect(canvases.size()).toBe(3, msg);
+        
+        canvases.each(function(element, index) {
+          
+            if (index == 0) { // FIXME: we assumed here that the context is the first item but may be not.
```

Comment:
Yes, the context _should_ always be the first one, if it's not maybe other tests will fail (not just this one). No need to for an additional check.